zkevm-circuits/src/blake2b/src/main.rs

```diff
@@ -0,0 +1,442 @@
+extern crate halo2_proofs;
+use halo2_proofs::{arithmetic::FieldExt, circuit::*, plonk::*, poly::Rotation, dev::MockProver, halo2curves::bn256::Fr};
+use std::{marker::PhantomData, array, convert::TryInto};
+
+fn gf<F:FieldExt>(n: u64) -> Expression<F> {
+    Expression::Constant(F::from(n))
+}
+
+fn array_refs_from_slice<T, const S: usize, const L: usize>(slice: &[T]) -> [&[T; S]; L] {
+    array::from_fn(|i| slice[S * i..][..S].try_into().unwrap())
+}
+
+fn byte_cells_to_number<F:FieldExt>(meta: &mut VirtualCells<F>, cells: &[ACell<F>]) -> Expression<F> {
+    cells.iter().map(|c| c.express(meta)).rfold(gf(0), |e, b| { e * gf(256) + b })
```

Comment:
```suggestion
    cells.iter().map(|cell| cell.expr(meta)).rfold(0.expr(), |value, part| { value * gf(256) + part })
```
or something like that I think improves readability a bit, but yes it is a bit longer.

 I see use a lot of single letter variable names elsewhere as well, making them a bit longer so they have some meaning I think is worth it.